### PR TITLE
Added max lines and overflow scrollbar for `gr.Code`

### PR DESCRIPTION
**Pull request by micpst**
## Description

Adds `max_lines` parameter for `gr.Code` component, similar to `gr.Textbox`. This can be used for controlling the display size of this component.

Please run `js/component-test` to see it in action (go to Code tab)
```sh
pnpm i
cd js/component-test
pnpm dev
```

I kindly request you @abidlabs to please review this change and let me know your feedback! 🙂

Related to #9192

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: gradio/components/code.py [hunk 1]
```diff
@@ -92,7 +92,8 @@ def __init__(
         *,
         every: Timer | float | None = None,
         inputs: Component | Sequence[Component] | set[Component] | None = None,
-        lines: int = 5,
+        lines: int = 1,
```
[the remark this answers is not in the document]
Comment (micpst) [reply gradio/components/code.py:95]: Just pure esthetic change, imho this looks better for one-liners, as we're not wasting space. I also noticed that the default `lines` param for `gr.Textbox` is also set to 1. I can revert this, no strong preference here.

[the remark this answers is not in the document]
Comment (abidlabs) [reply gradio/components/code.py:95]: In that case, would you kindly be able to rebase this PR and target the `5.0-dev` branch instead @micpst? That's the branch we are working on for the upcoming 5.0 release, and is more appropriate for any breaking changes

[the remark this answers is not in the document]
Comment (micpst) [reply gradio/components/code.py:95]: I'd love to have this functionality in the minor release. I can open another PR for the `5.0-dev` branch to change the default `lines` if you want.

[the remark this answers is not in the document]
Comment (abidlabs) [reply gradio/components/code.py:95]: Sure, I would say if you are targeting 4.x, then go ahead and revert this change

**Comment by abidlabs** [comment]
Btw @micpst we are temporarily not merging PRs to `main` while we prepare for our 5.0 release. In the meantime, you can use the wheel generated by this PR to install Gradio from this PR: 

```
pip install https://gradio-pypi-previews.s3.amazonaws.com/0792b5d0ed190e067d5ef438e4d2960597e7757e/gradio-4.44.0-py3-none-any.whl
```

but I'll probably rebase this to target `5.0-dev`

**Comment by abidlabs** [comment]
Hi @micpst I just tested this PR and it looks like the behavior of `max_lines` is correct, but `lines` is not behaving as expected. E.g. in this example:

```py
import gradio as gr

with gr.Blocks() as demo:
    gr.Code(lines=5)

demo.launch()
```

only 1 line is shown instead of 5:

image

**Comment by micpst** [comment]
Hi @abidlabs this is the standard behavior of the `lines` parameter, the current version of gradio renders the `Code` component in the same way.

**Comment by abidlabs** [comment]
Ah you're right. Tested and confirms this PR works correct. Brilliant, thanks @micpst!

**Pull request merged**